js/analyzer.js

```diff
@@ -53,94 +89,251 @@ Analyzer.prototype.abortQueue = function(){
     this.queue = [];
 };
 
-
 /**
- * checks the keyword density of given keyword against the cleantext stored in _store.
+ * checks the keyword density of given keyword against the cleantext stored in __store.
  * @returns resultObject
  */
-Analyzer.prototype.keywordDensity = function(){
-    if(yst_preProcessor._store.wordcount > 100) {
-        var keywordMatches = yst_preProcessor._store.cleanText.match(new RegExp(this.config.keyword, 'g'));
-        var keywordDensity = 0;
-        if ( keywordMatches !== null ) {
-            var keywordCount = keywordMatches.length;
-            var keywordDensity = (keywordCount / yst_preProcessor._store.wordcount - (keywordCount - 1 * keywordCount)) * 100;
-        }
-        switch (true) {
-            case (keywordDensity < 1):
-                var rating =  4;
-                break;
-            case (keywordDensity > 4.5):
-                var rating = -50;
-                break;
-            default:
-                var rating = 9;
-                break;
-        }
-        result = {name: 'keywordDensity', result: {keywordDensity: keywordDensity.toFixed(1) }, rating: rating};
-    }else{
-        result = {name: 'keywordDensity', result: null, rating: null};
+Analyzer.prototype.keywordDensity = function() {
+    if (yst_preProcessor.__store.wordcount > 100) {
+        var keywordDensity = this.keywordDensityCheck();
+        var rating = this.keywordDensityRating(keywordDensity);
+        return {name: "keywordDensity", result: {keywordDensity: keywordDensity.toFixed(1)}, rating: rating};
+    } else {
```

Comment:
In PHP I would remove the else and return the value within this else directly, because the if already break out the method. Is the same possible in JavaScript?